agent/reconciler/host_reconciler.go

```diff
@@ -302,3 +260,67 @@ func (r *HostReconciler) installK8sComponents(ctx context.Context, byoHost *infr
 	conditions.MarkTrue(byoHost, infrastructurev1beta1.K8sComponentsInstallationSucceeded)
 	return nil
 }
+
+func (r *HostReconciler) uninstallk8sComponents(ctx context.Context, byoHost *infrastructurev1beta1.ByoHost) error {
+	logger := ctrl.LoggerFrom(ctx)
+	
+	bundleRegistry := byoHost.GetAnnotations()[infrastructurev1beta1.BundleLookupBaseRegistryAnnotation]
+	k8sVersion := byoHost.GetAnnotations()[infrastructurev1beta1.K8sVersionAnnotation]
+	byohBundleTag := byoHost.GetAnnotations()[infrastructurev1beta1.BundleLookupTagAnnotation]
+	bundleInstaller, err := installer.New(bundleRegistry, r.DownloadPath, logger)
+	if err != nil {
+		return err
+	}
+	err = bundleInstaller.Uninstall(k8sVersion, byohBundleTag)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (r *HostReconciler) removeAnnotations(ctx context.Context, byoHost *infrastructurev1beta1.ByoHost) error {
+	logger := ctrl.LoggerFrom(ctx)
+	logger.Info("Removing the bootstrap sentinel file...")
+	if _, err := os.Stat(bootstrapSentinelFile); !os.IsNotExist(err) {
+		err := os.Remove(bootstrapSentinelFile)
+		if err != nil {
+			return errors.Wrapf(err, "failed to delete sentinel file %s", bootstrapSentinelFile)
+		}
+	}
+	if IP, ok := byoHost.Annotations[infrastructurev1beta1.EndPointIPAnnotation]; ok {
+		network, err := vip.NewConfig(IP, registration.LocalHostRegistrar.ByoHostInfo.DefaultNetworkInterfaceName, false)
+		if err == nil {
+			err := network.DeleteIP()
+			if err != nil {
+				return err
+			}
+		}
+	}
```

Comment:
Will it better if we rename this to `removeMetadata` or `removeClusterMetadata`?

Reply:
`removeMetadata` sounds okay to me. @anusha94 thoughts??

Reply:
> This code doesn't align with the method name removeAnnotations

+1

I would not mind having 2 more functions - `removeSentinelFile()` and `deleteEndpointIP()` that will be called from `hostCleanUp` function.
Or you can choose to keep this inside `hostCleanUp` itself like @dharmjit suggested.

Reply:
`removeAnnotations` is a better name IMO - for annotations related stuff

Reply:
I think adding two more functions `removeSentinelFile()` and `deleteEndpointIP()` along with `removeAnnotations()` sounds good. It would keep the `hostCleanup` function clean